compute/client_library/snippets/disks/resize_disk.py

```diff
@@ -0,0 +1,120 @@
+    they will be printed to sys.stderr.
+
+    Args:
+        operation: a long-running operation you want to wait on.
+        verbose_name: (optional) a more verbose name of the operation,
+            used only during error and warning reporting.
+        timeout: how long (in seconds) to wait for operation to finish.
+            If None, wait indefinitely.
+
+    Returns:
+        Whatever the operation.result() returns.
+
+    Raises:
+        This method will raise the exception received from `operation.exception()`
+        or RuntimeError if there is no exception set, but there is an `error_code`
+        set for the `operation`.
+
+        In case of an operation taking longer than `timeout` seconds to complete,
+        a `concurrent.futures.TimeoutError` will be raised.
+    """
+    result = operation.result(timeout=timeout)
+
+    if operation.error_code:
+        print(
+            f"Error during {verbose_name}: [Code: {operation.error_code}]: {operation.error_message}",
+            file=sys.stderr,
+            flush=True,
+        )
+        print(f"Operation ID: {operation.name}", file=sys.stderr, flush=True)
+        raise operation.exception() or RuntimeError(operation.error_message)
+
+    if operation.warnings:
+        print(f"Warnings during {verbose_name}:\n", file=sys.stderr, flush=True)
+        for warning in operation.warnings:
+            print(f" - {warning.code}: {warning.message}", file=sys.stderr, flush=True)
+
+    return result
+
+
+def resize_disk(project_id: str, disk_link: str, new_size_gb: int) -> NoReturn:
+    """
+    Resizes a persistent disk to a specified size in GB. After you resize the disk, you must
+    also resize the file system so that the operating system can access the additional space.
+
+    Args:
+        project_id:
+        disk_link:
+        new_size_gb:
+    """
+    search_results = re.search(
+        r"/projects/[\w_-]+/(?P<area_type>zones|regions)/"
+        r"(?P<area_name>[\w_-]+)/disks/(?P<disk_name>[\w_-]+)",
+        disk_link,
+    )
+
+    if search_results["area_type"] == "regions":
+        disk_client = compute_v1.RegionDisksClient()
+        request = compute_v1.ResizeRegionDiskRequest()
+        request.region = search_results["area_name"]
+        request.region_disks_resize_request_resource = (
+            compute_v1.RegionDisksResizeRequest()
+        )
+        request.region_disks_resize_request_resource.size_gb = new_size_gb
+    else:
+        disk_client = compute_v1.DisksClient()
+        request = compute_v1.ResizeDiskRequest()
+        request.zone = search_results["area_name"]
+        request.disks_resize_request_resource = compute_v1.DisksResizeRequest()
+        request.disks_resize_request_resource.size_gb = new_size_gb
+
+    request.disk = search_results["disk_name"]
+    request.project = project_id
+
+    operation = disk_client.resize(request)
+    wait_for_extended_operation(operation, "disk resize")
+
+    return
```

Comment:
Is explicitly including the return statement a Python sample best practice?

Either way, can we move the region tag end statements up to avoid the extra blank lines in the doc sample?